api/pacman-api-admin/src/main/java/com/tmobile/pacman/api/admin/controller/AccountsController.java

```diff
@@ -217,40 +208,48 @@ public ResponseEntity<Object> deletePlugin(@ApiParam(value = "provide account nu
-            PluginResponse response = redHatPluginService.createPlugin(request, createdBy);
-            if (response.getStatus().equals(AdminConstants.SUCCESS)) {
-                List<String> optionalAssetList = null;
-                if (Objects.nonNull(optionalAssetGroupList)) {
-                    optionalAssetList = Arrays.asList(optionalAssetGroupList.split(","));
-                }
-                if (Objects.nonNull(optionalAssetList) && !optionalAssetList.isEmpty()
-                        && optionalAssetList.contains(type)) {
-                    String updateAssetGroupStatus = assetGroupService.updateAssetGroupStatus(type, true, "admin");
-                    log.info("AssetGroup  {} status {}", type, updateAssetGroupStatus);
-                }
-                log.info(type + " plugin created successfully.");
+            PluginParameters parameters = PluginParameters.builder().pluginName(type).createdBy(createdBy).build();
+            if (type.equals("redhat")) {
+                request = objectMapper.convertValue(request, RedHatPluginRequest.class);
+                response = redHatPluginService.createPlugin((RedHatPluginRequest) request, parameters);
+            } else if (type.equals("gcp")) {
+                request = objectMapper.convertValue(request, GcpPluginRequest.class);
+                response = gcpPluginService.createPlugin((GcpPluginRequest) request, parameters);
             } else {
-                log.info("Failed to create " + type + " plugin.");
+                return ResponseUtils.buildFailureResponse(new Exception(UNEXPECTED_ERROR_OCCURRED),
+                        "Invalid request body or plugin type");
             }
             return ResponseUtils.buildSucessResponse(response);
+        } catch (PluginServiceException pse) {
+            log.error(pse.getMessage(), pse);
+            return ResponseUtils.buildFailureResponse(pse, pse.getMessage());
         } catch (Exception exception) {
             log.error(UNEXPECTED_ERROR_OCCURRED, exception);
-            if (redHatPluginService != null) {
-                redHatPluginService.deletePlugin(request.getRedhatAccountId());
-            }
             return ResponseUtils.buildFailureResponse(new Exception(UNEXPECTED_ERROR_OCCURRED), exception.getMessage());
         }
     }
 
     @ApiOperation(httpMethod = "POST", value = "API to validate plugin configuration", response = Response.class, produces = MediaType.APPLICATION_JSON_VALUE)
     @PostMapping(path = "/{type}/validate", produces = MediaType.APPLICATION_JSON_VALUE)
-    public ResponseEntity<Object> validateAccount(@RequestBody RedHatPluginRequest request, @PathVariable("type") String type) {
+    public ResponseEntity<Object> validateAccount(@RequestBody Object request, @PathVariable("type") String type) {
         try {
-            return ResponseUtils.buildSucessResponse(redHatPluginService.validate(request));
+            if (type.equals("redhat")) {
```

Comment:
Same here, we can think of a way to generalize this.

The reason being is that we will be release many plugins that follow a very similar template, hence we can think for plugin #3 (contrast for example) and do some refactoring.